native/libp2p_port/internal/subscriptions/subscriptions.go

```diff
@@ -21,9 +22,12 @@ type subscription struct {
 }
 
 type Subscriber struct {
-	subscriptions map[string]subscription
-	gsub          *pubsub.PubSub
-	port          *port.Port
+	// NOTE: since subscriptions is accessed only for topic un/subscription
+	// and message publishing, this lock doesn't have much contention
+	subscriptionsMutex *sync.Mutex
```

Comment:
When would we need a mutex? AFAIK commands are handled in a sequence, so we wouldn't ever subscribe or unsubscribe concurrently.

Reply:
If we want to handle commands concurrently (a fairly reasonable thing to do) in goroutines, then we might want to add a separate issue for that.

Reply:
You're right. Before it was needed, but I think I fixed it when refactoring. I'll remove it.